azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -139,26 +132,28 @@ def update(self, force_update=False):
         """
         Updates the current GoalState instance fetching values from the WireServer/HostGAPlugin as needed
         """
-        xml_text, xml_doc = GoalState._fetch_goal_state(self._wire_client)
-
-        vm_settings = self._fetch_vm_settings(force_update=force_update)
+        timestamp = create_timestamp()
+        xml_text, xml_doc, incarnation = GoalState._fetch_goal_state(self._wire_client)
```

Comment:
where/when is this full goal state saved to history folder?

Reply:
it's be saved as part of _fetch_extended_goal_state (called below @line 142) or after retrieving the vmsettings (below @line 150)